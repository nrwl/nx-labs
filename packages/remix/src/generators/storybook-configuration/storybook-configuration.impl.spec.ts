import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import libraryGenerator from '../library/library.impl';
import storybookConfigurationGenerator from './storybook-configuration.impl';

describe('Storybook Configuration', () => {
  it.each(['jest', 'vitest', 'none'])(
    'it should create a storybook configuration and use react-vite framework with testing framework %s',
    async (unitTestRunner: 'jest' | 'vitest' | 'none') => {
      // ARRANGE
      const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });

      await libraryGenerator(tree, {
        name: 'storybook-test',
        style: 'css',
        unitTestRunner,
      });

      // ACT
      await storybookConfigurationGenerator(tree, {
        name: 'storybook-test',
        configureCypress: false,
        configureStaticServe: false,
        generateStories: true,
      });

      // ASSERT
      expect(tree.exists(`libs/storybook-test/vite.config.ts`));
      expect(
        tree.read(`libs/storybook-test/.storybook/main.js`, 'utf-8')
      ).toMatchSnapshot();
    }
  );
});
