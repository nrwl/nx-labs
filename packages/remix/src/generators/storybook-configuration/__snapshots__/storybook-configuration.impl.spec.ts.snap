// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storybook Configuration it should create a storybook configuration and use react-vite framework with testing framework jest 1`] = `
"const config = {
  stories: ['../src/lib/**/*.stories.@(js|jsx|ts|tsx|mdx)'],
  addons: ['@storybook/addon-essentials'],
  framework: {
    name: '@storybook/react-vite',
    options: {
      builder: {
        viteConfigPath: '',
      },
    },
  },
};

export default config;

// To customize your Vite configuration you can use the viteFinal field.
// Check https://storybook.js.org/docs/react/builders/vite#configuration
// and https://nx.dev/packages/storybook/documents/custom-builder-configs
"
`;

exports[`Storybook Configuration it should create a storybook configuration and use react-vite framework with testing framework none 1`] = `
"const config = {
  stories: ['../src/lib/**/*.stories.@(js|jsx|ts|tsx|mdx)'],
  addons: ['@storybook/addon-essentials'],
  framework: {
    name: '@storybook/react-vite',
    options: {
      builder: {
        viteConfigPath: '',
      },
    },
  },
};

export default config;

// To customize your Vite configuration you can use the viteFinal field.
// Check https://storybook.js.org/docs/react/builders/vite#configuration
// and https://nx.dev/packages/storybook/documents/custom-builder-configs
"
`;

exports[`Storybook Configuration it should create a storybook configuration and use react-vite framework with testing framework vitest 1`] = `
"const config = {
  stories: ['../src/lib/**/*.stories.@(js|jsx|ts|tsx|mdx)'],
  addons: ['@storybook/addon-essentials'],
  framework: {
    name: '@storybook/react-vite',
    options: {
      builder: {
        viteConfigPath: '',
      },
    },
  },
};

export default config;

// To customize your Vite configuration you can use the viteFinal field.
// Check https://storybook.js.org/docs/react/builders/vite#configuration
// and https://nx.dev/packages/storybook/documents/custom-builder-configs
"
`;
